framework/src/modules/chain/blocks/utils.js

```diff
@@ -112,66 +31,44 @@ const parseStorageObjToLegacyObj = block => {
  * @returns {Object} cb.rows - List of blocks
  */
 // eslint-disable-next-line class-methods-use-this
-const loadBlocksDataWS = async (storage, filter, tx) => {
-	const params = { limit: filter.limit || 1 };
-
-	// FIXME: filter.id is not used
-	if (filter.id && filter.lastId) {
-		throw new Error('Invalid filter: Received both id and lastId');
+const loadBlocksFromLastBlockId = async (storage, lastBlockId, limit) => {
+	if (!lastBlockId) {
+		throw new Error('lastBlockId needs to be specified');
 	}
-	if (filter.id) {
-		params.id = filter.id;
-	} else if (filter.lastId) {
-		params.lastId = filter.lastId;
+	if (!limit) {
```

Comment:
What about add a default value and remove this check?

Reply:
actually it is provided in the calling function. maybe we don't need this check at all, but it was just for safety